src/data/make_dataset.py

```diff
@@ -0,0 +1,75 @@
+"""
+make_dataset.py
+
+This module contains functions to process raw data and create a final dataset ready for analysis.
+"""
+# -*- coding: utf-8 -*-
+import os
+import pickle
+import sys
+import numpy as np
+from sklearn.preprocessing import LabelEncoder
+from keras._tf_keras.keras.preprocessing.text import Tokenizer
+from keras._tf_keras.keras.preprocessing.sequence import pad_sequences
+
+
+def read_data(data_file):
+    """
+    Processes raw data and creates a final dataset ready for analysis.
+    """
+    with open(data_file, "r", encoding="utf-8") as f:
+        data_lines = f.readlines()[1:]
+
+    data = [line.strip() for line in data_lines]
+    raw_x_data = [line.split("\t")[1] for line in data]
+    raw_y_data = [line.split("\t")[0] for line in data]
+
+    return raw_x_data, raw_y_data
+
+
+def tokenize_data(raw_x_train, raw_x_val, raw_x_test, raw_y_train, raw_y_val, raw_y_test):
```

Comment:
this method is now in the lib-ml library, you need to make us of it instead of having the logic here https://github.com/remla2024-team11/lib-ml/blob/main/src/lib_ml_team11/preprocessing.py#L24